gateway/observation.go

```diff
@@ -0,0 +1,66 @@
+package gateway
+
+import (
+	"net/http"
+
+	"github.com/google/uuid"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/buildwithgrove/path/observation"
+)
+
+const (
+	httpHeaderAuthRequestID       = "Auth-Request-ID"
+	httpHeaderAuthRegion          = "Auth-Region"
+	httpHeaderAuthPortalAccountID = "Auth-Portal-Account-ID"
```

Comment:
Ditto my comment in `auth.proto`

- Rename `RequestId` appropriately
- Decouple portal related things